FHIR/sushi
# Extension contains syntax

Instead of treating slice name as both the slice name and a potential reference to an external extension, make an explicit syntax for declaring the slice name *and* type.  E.g.,
```
  Alias: MaxSizeExtension: http://hl7.org/fhir/StructureDefinition/maxSize
  // then within a profile definition:
  extension contains MaxSizeExtension named maxSize
```
See conversation: https://chat.fhir.org/#narrow/stream/215610-shorthand/topic/Proposed.20Syntax.20Change.20for.20Extensions

Fixes #86 and #182.

**TO TEST**:
1. Run this branch against `fsh-mcode` `master` branch
2. Note the warnings; and note that the 1st warning is different than the others, because it concerns an extension that is defined outside the FSH Tank
3. Run this branch against `fsh-mcode` `new-extension-syntax` branch and note that the warnings go away
4. (Optional) Look at [new-extension-syntax](https://github.com/standardhealth/fsh-mcode/compare/new-extension-syntax) diff to see changes that resulted in removing warnings
5. Compare output of `sushi:master`/`fsh-mcode:master` run with `sushi:contains-extension-syntax`/`fsh-mcode:new-extension-syntax` run.  The results should be identical (well, there are two diffs in a *description* because I trimmed trailing whitespace off the description).
